core/src/machine/step.rs

```diff
@@ -433,12 +583,59 @@ impl Machine {
                 }
             }
 
+            // ========== JP ==========
+            opcode!("JP a16") => self.cpu.pc = arg_word,
+            opcode!("JP (HL)") => self.cpu.pc = self.cpu.hl(),
```

Comment:
Mh I think the mnemonic is misleading here (not the fault of this PR). I would change it to `JP HL`. It not called `JP (a16)` either. And [this page](http://gbdev.gg8.se/wiki/articles/CPU_Instruction_Set) also calls it `JP HL`. Maybe you could change that in this PR still (including `instr.rs`, of course).